src/simulation/m_global_parameters.fpp

```diff
@@ -612,7 +692,7 @@ contains
                     ! print*, 'alf idx', alf_idx
                     ! print*, 'bub -idx beg end', bub_idx%beg, bub_idx%end
 
-                    @:ALLOCATE(weight(nb), R0(nb), V0(nb))
+                    @:ALLOCATE_GLOBAL(weight(nb), R0(nb), V0(nb))
```

Comment:
is this correct for nvhpc?

Reply:
For nvhpc (and CCE 17, once I verify we can turn off CRAY_ACC_WAR) `ALLOCATE_GLOBAL` and `ALLOCATE` should be identical. I'll skim the standard to see if there's reason you may want to differentiate between module allocatables and locally scoped allocatables in the future, besides the general fragility of FORTRAN module variables in directive languages.

Reply:
good to know. there is a bug causing this code to not run on nvhpc + Nvidia GPUs at the moment, so I was checking out the changes to see what might be an issue.

Reply:
Yes, I have rewritten all the macros so that it does the same thing as before